File: leetcode/easy/495-teemoAttacking.java
// first pass solution
class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int total = 0;
        
        for (int i = 0; i < timeSeries.length - 1; i++) {
            if (timeSeries[i] + duration < timeSeries[i+1]) {
                total += duration;
            } else {
                total += timeSeries[i+1] - timeSeries[i];
            }
        }
        return total + duration;
    }
}

File: leetcode/easy/495-teemoAttacking.cpp
class Solution {
public:
    int findPoisonedDuration(vector<int>& timeSeries, int duration) {
        int output = timeSeries.size() > 0 ? duration : 0;

        for (int i = 1; i < timeSeries.size(); i++) {
            if (timeSeries[i] > timeSeries[i-1] + duration) {
                output += duration;
            } else {
                output += (timeSeries[i] + duration) - (timeSeries[i-1] + duration);
            }
        }

        return output;
    }
};
